packages/components/alert/src/alert.vue

```diff
@@ -91,6 +92,16 @@ const { onOpen, onClose } = useDelayedToggle({
 })
 
 if (isClient) {
-  onOpen()
+  const rawProps = getCurrentInstance()?.vnode?.props ?? {}
+
+  if (
+    !['showAfter', 'show-after', 'show-After'].some((key) =>
+      hasOwn(rawProps, key)
+    )
+  ) {
+    open()
+  } else {
+    onOpen()
+  }
```

Comment:
We can inverse the condition for readability.

```suggestion

  if (
    ['showAfter', 'show-after', 'show-After'].some((key) =>
      hasOwn(rawProps, key)
    )
  ) {
    onOpen()
  } else {
    open()
  }
```